packages/flask/src/UIAnimatedBalance/UIAnimatedBalance.tsx

```diff
@@ -0,0 +1,286 @@
+                    prevState.integer.length + prevState.fractional.length
+            ) {
+                return;
+            }
+
+            const numRegExp = /[0-9]/g;
+            const wideNumLiteral = '5';
+
+            normalized.value =
+                state.integer.replace(numRegExp, wideNumLiteral) +
+                decimalSeparator +
+                state.fractional.replace(numRegExp, wideNumLiteral);
+        },
+    );
+
+    /**
+     * Basically we exploit the fact that TextView's text
+     * can be updated with `setNativeProps`,
+     * in order to not do it manually just delegating it here
+     * to reanimated, that will do the same under the hood
+     */
+    const animatedIntegerProps: any = useAnimatedProps(() => {
+        return {
+            text: formatted.value.integer,
+        };
+    });
+    const animatedFractionalProps: any = useAnimatedProps(() => {
+        return {
+            text: formatted.value.fractional,
+        };
+    });
+
+    const normalizedProps: any = useAnimatedProps(() => {
+        return {
+            text: normalized.value,
+        };
+    });
+
+    const iconElement = useIcon(icon, integerVariant, loading);
+
+    const colorStyle = React.useMemo(() => ({ color: theme[ColorVariants.TextPrimary] }), [theme]);
+
+    const textLikeContainer = React.useMemo(
+        () => (I18nManager.isRTL ? styles.rtlContainer : styles.ltrContainer),
+        [I18nManager.isRTL],
+    );
+
+    return (
+        <View style={styles.intrinsicWrapper}>
+            <View style={[textLikeContainer, styles.main]}>
+                <AnimatedTextInput
+                    style={[Typography[integerVariant], styles.hidden]}
+                    animatedProps={normalizedProps}
+                    defaultValue={normalized.value}
+                    underlineColorAndroid="transparent"
+                    editable={false}
+                />
+                {/*
+                 * Space letter here is important here, it's not a mistake!
+                 * It's placed to have a separation from main balance,
+                 * instead of `marginLeft` as space is more "text friendly"
+                 * and will layout properly in text
+                 */}
+                <Text style={Typography[fractionalVariant]}> {iconElement}</Text>
+                <View style={[textLikeContainer, styles.visible]}>
+                    <AnimatedTextInput
+                        style={[Typography[integerVariant], colorStyle, styles.integer]}
+                        animatedProps={animatedIntegerProps}
+                        defaultValue={formatted.value.integer}
+                        underlineColorAndroid="transparent"
+                        editable={false}
+                    />
+                    <Text style={[Typography[fractionalVariant], colorStyle, styles.delimeter]}>
+                        {decimalSeparator}
+                    </Text>
```

Comment:
In my opinion, the separator should not be displayed if `maxFractionalDigits === 0`

Reply:
Good point 🤔 I'm just thinking that what the point to show balance in blockchain when there is no fractional digits? we actually not making general thing to show any number, so to make it simpler, what if imply that there should be some amount of fractional digits? ie set type of `maxFractionalDigits` to sth like `1 | 2 | 3 | 4 | 5`, that would also eliminate case when someone pass huge amount of fractional digits, that would be filled with 0

Reply:
Maybe you're right. But I would be reinsured, and worked out this moment. We do not know where and for what purposes else this component can be used. It is possible that integer precision will be enough for someone or sometime for us.

Reply:
For us, I know exact places where it's used, and it has 2 fractional digits in both cases (Initially I even thought to make it exactly 2, but then decided to give a bit of flexibility with a prop as it was before (and it wasn't used actually)). It's actually designed for this particular place, and to use it in different context may require another design, and fractional count might be the lesser problem, so I vote to solve problems as they appear :)

Reply:
I don't mind :)